go/integrityblock/integrityblock.go

```diff
@@ -33,22 +33,30 @@ var IntegrityBlockMagic = []byte{0xf0, 0x9f, 0x96, 0x8b, 0xf0, 0x9f, 0x93, 0xa6}
 // "b1" as bytes and 2 empty bytes
 var VersionB1 = []byte{0x31, 0x62, 0x00, 0x00}
 
-// CborBytes returns the CBOR encoded bytes of an integrity signature.
-func (is *IntegritySignature) CborBytes() ([]byte, error) {
-	var buf bytes.Buffer
-	enc := cbor.NewEncoder(&buf)
-	enc.EncodeArrayHeader(2)
-
+// cborEncodeSignatureAttributesMap writes the signature attributes map as CBOR using the given encoder so that the map's key is text string and value byte string.
+func cborEncodeSignatureAttributesMap(signatureAttributes map[string][]byte, enc *cbor.Encoder) error {
 	mes := []*cbor.MapEntryEncoder{}
-	for key, value := range is.SignatureAttributes {
+	for key, value := range signatureAttributes {
 		mes = append(mes,
 			cbor.GenerateMapEntry(func(keyE *cbor.Encoder, valueE *cbor.Encoder) {
 				keyE.EncodeTextString(key)
 				valueE.EncodeByteString(value)
 			}))
 	}
 	if err := enc.EncodeMap(mes); err != nil {
-		return nil, fmt.Errorf("integrityblock: Failed to encode signature attribute: %v", err)
+		return fmt.Errorf("integrityblock: Failed to encode signature attributes: %v", err)
+	}
+	return nil
+}
+
+// CborBytes returns the CBOR encoded bytes of an integrity signature.
+func (is *IntegritySignature) CborBytes() ([]byte, error) {
```

Comment:
Can we also make this function take a `*cbor.Encoder` so that `IntegrityBlock.CborBytes()` can pass its encoder?